### PR TITLE
[Merged by Bors] - [Events] Add `limit` option back

### PR DESCRIPTION
**Pull request by nlordell**
The limit option was removed during a refactor, this PR adds it back.

### Test Plan

New unit test to make sure it is properly being added to the logs query.

### CODE REVIEW

**Review by e00E — APPROVED** [review]

File: src/contract/event.rs [hunk 1]
```diff
@@ -229,6 +237,14 @@ impl<T: Transport, E: ParseLog> AllEventsBuilder<T, E> {
         self
     }
 
+    /// Limit the number of events that can be retrieved by this filter.
+    ///
+    /// Note that this parameter is non-standard.
+    pub fn limit(mut self, value: usize) -> Self {
+        self.filter = self.filter.limit(value);
+        self
+    }
```
Comment (bh2smith) [inline src/contract/event.rs:246]: I see this snippet in more than 1 place... Is this intended?

**Review by bh2smith — APPROVED** [review]
Looks good. I only wonder why was it ever removed in the first place?

File: src/contract/event.rs (hunk 1 above)
Comment (nlordell) [reply src/contract/event.rs:246]: > I only wonder why was it ever removed in the first place?

It was removed as part of a refactor, with intention of adding it back later.

> Is this intended?

Yes, there are more than one kind of builder that can set the `limit` property. These methods are just small helpers for setting that property.

**Pull request closed without merging**